src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -882,20 +885,18 @@ static BOOL AttributeHasChangesOmittedQuality(MTRAttributePath * attributePath)
         NSArray * readRequestData = @[ readRequestPath, params ?: [NSNull null] ];
 
         // But first, check if a duplicate read request is already queued and return
-        if ([_asyncCallbackWorkQueue isDuplicateForTypeID:MTRDeviceWorkItemDuplicateReadTypeID workItemData:readRequestData]) {
+        if ([_asyncWorkQueue hasDuplicateForTypeID:MTRDeviceWorkItemDuplicateReadTypeID workItemData:readRequestData]) {
             return attributeValueToReturn;
         }
 
         NSMutableArray<NSArray *> * readRequests = [NSMutableArray arrayWithObject:readRequestData];
 
         // Create work item, set ready handler to perform task, then enqueue the work
         MTRAsyncWorkItem * workItem = [[MTRAsyncWorkItem alloc] initWithQueue:self.queue];
-        MTRAsyncWorkBatchingHandler batchingHandler = ^(id opaqueDataCurrent, id opaqueDataNext, BOOL * fullyMerged) {
+        [workItem setBatchingID:MTRDeviceWorkItemBatchingReadID data:readRequests handler:^(id opaqueDataCurrent, id opaqueDataNext, BOOL * fullyMerged) {
             NSMutableArray<NSArray *> * readRequestsCurrent = opaqueDataCurrent;
             NSMutableArray<NSArray *> * readRequestsNext = opaqueDataNext;
 
-            *fullyMerged = NO;
```

Comment:
It’s pretty standard to have output parameters like `stop` etc initialised by the caller to default values. I think relying on every block instance to do this is more problematic. I would probably prefer if this was a return value instead of an out parameter…

Reply:
I am not sure why it wasn't made a return value to start with, though @jtung-apple might recall.  Maybe a followup to change that?

In any case, I am OK with the caller pre-initializing to NO, but needs to be documented.

Reply:
I didn't make these return values because the meaning of the return value wasn't clear from the name of these handlers. At the time I felt that this is the clearest way to have code self-document.

I'm not opposed to having it as return value, but I think it needs to be documented well. Either have the handler names be named sufficiently well to imply the meaning of the return, or have it return an enum, which can be clearer than this.

Reply:
I'll add some explicit comments, but I think it's fairly obvious from when the comment says "to indicate such and such set `*foo` to YES" that it starts out as NO. See the `stop` parameter in https://developer.apple.com/documentation/foundation/nsarray/1415846-enumerateobjectsusingblock for an example.